pkg/apiserver/validator.go

```diff
@@ -109,7 +102,7 @@ func (v *validator) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		} else {
 			errorMsg = "nil"
 		}
-		reply = append(reply, ServerStatus{name, status.String(), status, msg, errorMsg})
+		reply = append(reply, api.ServerStatus{name, status.String(), status, msg, errorMsg})
```

Comment:
That's not the versioned object.

It seems like a good idea to version this API, but it needs to change pretty significantly to make that possible.

For instance, we need to install version-specific handlers at version-specific paths.

The current InstallValidator:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/apiserver/apiserver.go#L143

probably needs to move to:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/apiserver/api_installer.go

However, since Openshift also uses apiserver and Kubelet will use it in the future, it can't just be installed by default. 

This may be the first example where we really need a separate API prefix #3806, with separate types.go files #1744.

A simpler approach for now would be to not version this API yet:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/unversioned.go

That would prohibit breaking changes post-1.0, but we could expose new resources, if necessary.

We should rename this one to /componentstatuses, if that's what it is.

Also, this needs to use go-restful, so that the documentation shows up in Swagger.

cc @smarterclayton, in case he has preferences/requirements or suggestions.